analyzers/src/SonarAnalyzer.CSharp/Rules/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -27,14 +27,15 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Extensions;
 using SonarAnalyzer.Helpers;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     public sealed class UnchangedLocalVariablesShouldBeConst : SonarDiagnosticAnalyzer
     {
         private const string DiagnosticId = "S3353";
-        private const string MessageFormat = "Add the 'const' modifier to '{0}'.";
+        private const string MessageFormat = "Add the 'const' modifier to '{0}'{1}.";
```

Comment:
I suggest to move `$", and replace 'var' with` here

```suggestion
        private const string MessageFormat = "Add the 'const' modifier to '{0}', and replace 'var' with {1}.";
```

Reply:
The `, and replace 'var' with {1}` part is conditional. We can either use `{1}` for the whole part or use a second `MessageFormat`.

Reply:
I'd prefer a second message format, otherwise it's not very clear

Reply:
With a second message, we need to rewrite `SupportedDiagnostics` too. `messageFormat` is mandatory for the DiagnosticDescriptor. Therefore we need two `Rule` properties and return only one of these in `SupportedDiagnostics`. That seems unfortunate to me, or do I miss something?

```cs
        private const string MessageFormat = "Add the 'const' modifier to '{0}'{1}.";
        private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
```

Reply:
I feel an itch that the messages are in two different parts of the code, maybe if the second optional part would be under MessageFormat my itch would go away

Or like: https://github.com/SonarSource/sonar-dotnet/blob/b4b5df2265374be44123d99afde8067ed99358f2/analyzers/src/SonarAnalyzer.CSharp/Rules/BinaryOperationWithIdenticalExpressions.cs#L35

```DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, "{0}");```

and have two message formats